### PR TITLE
Add Kinesis publisher channel type

### PR DESCRIPTION
**Pull request by mattfont**

### CODE REVIEW

File: chans/kdspub/kds_test.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2021 Comcast Cable Communications Management, LLC
```
Comment (jsccast) [inline chans/kdspub/kds_test.go:2]: Please change year to 2023.

File: chans/kdspub/kdspub.go [hunk 2]
```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2021 Comcast Cable Communications Management, LLC
```
Comment (jsccast) [inline chans/kdspub/kdspub.go:2]: Please change year to 2023.

File: chans/kdspub/kdspub.go [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+package kdspub
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/Comcast/plax/dsl"
+	"github.com/aws/aws-sdk-go-v2/aws"
```
Comment (jsccast) [inline chans/kdspub/kdspub.go:27]: Please add a newline between the `plax` import and the AWS SDK imports. Having multiple blocks like that improves readability (and that's the only benefit AFAIK).

File: chans/kdspub/kdspub.go [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+func (c *KDSPubChan) Sub(ctx *dsl.Ctx, topic string) error {
+	return dsl.Brokenf("Can't Sub on a KDS (%s)", c.opts.StreamName)
+}
+
+func (c *KDSPubChan) Pub(ctx *dsl.Ctx, m dsl.Msg) error {
+
+	ctx.Logf("Publishing to KDS %s", c.opts.StreamName)
+
+	// Load the Shared AWS Configuration (~/.aws/config)
```
Comment (jsccast) [inline chans/kdspub/kdspub.go:110]: This configuration and client creation probably belongs in `Open`. The SQS channel type [creates the session in `Open`](https://github.com/Comcast/plax/blob/main/chans/sqs/sqs.go#L140-L149), which seems fine. Maybe just do the same here.

**Review by jsccast — CHANGES_REQUESTED** [review]
Thanks for the PR. Just a few small changes, please. See the review comments.

Looks like `.DS_Store` snuck into the commit. Please remove that file.

Thanks again for the contribution.

File: chans/kdspub/kds_test.go (hunk 1 above)
Comment (mattfont) [reply chans/kdspub/kds_test.go]: Dome

File: chans/kdspub/kdspub.go (hunk 2 above)
Comment (mattfont) [reply chans/kdspub/kdspub.go]: Done

File: chans/kdspub/kdspub.go (hunk 3 above)
Comment (mattfont) [reply chans/kdspub/kdspub.go:28]: Done

File: chans/kdspub/kdspub.go (hunk 4 above)
Comment (mattfont) [reply chans/kdspub/kdspub.go:116]: Made the changes to mimic SQS channel.
Done

**Review by jsccast — APPROVED** [review]
Thanks!

**Pull request merged**